forge/csrc/forge_passes.cpp

```diff
@@ -230,116 +229,4 @@ graphlib::Graph *run_pre_lowering_passes(graphlib::Graph *graph, const std::opti
 
     return graph;
 }
-
-// ********** Run lowering passes **********
-std::unique_ptr<graphlib::Graph> run_pre_placer_forge_passes(
-    graphlib::Graph *graph,
-    const DeviceConfig &device_config,
-    std::vector<std::uint32_t> chip_ids,
-    const std::vector<std::string> &op_names_dont_fuse,
-    const std::vector<std::string> &op_names_manual_fuse,
-    const passes::FractureChipIdAssignments &fracture_chip_id_assignments,
-    const std::optional<DataFormat> default_df_override,
-    const std::optional<DataFormat> default_accumulate_df,
-    const bool enable_broadcast_splitting,
-    const DataFormat fp32_fallback,
-    const MathFidelity default_math_fidelity,
-    const bool enable_auto_fusing,
-    const int amp_level,
-    const bool enable_recompute,
-    const bool output_queues_on_host,
-    const bool input_queues_on_host,
-    const std::vector<std::tuple<std::string, std::string, int>> &insert_queues,
-    std::vector<AMPNodeProperties> amp_properties,
-    const std::vector<std::string> &op_intermediates_to_save,
-    const bool use_interactive_placer,
-    bool enable_device_tilize)
-{
-    log_debug(LogGraphCompiler, "Lowering target device\n{}", device_config);
-
-    passes::print_graph(graph, "PRE_PLACER");
-
-    // Create forge ops / tms
-    std::unique_ptr<graphlib::Graph> lowered_graph = lower_to_forge_ops(graph);
-
-    // lower user-defined buffering queues to actual queue types
-    lower_to_buffering_queues(lowered_graph.get());
-
-    split_unsupported_gradient_ops(lowered_graph.get(), device_config);
-    recalculate_shapes(lowered_graph.get());
-
-    // Remove nops
-    remove_nops(lowered_graph.get());
```

Comment:
Note: Remove NOPs pass is useful, as we're not lowering NOPs in MLIR :))

However, I'm more for deleting this one as well, and actually push effort on cleaning up NOP usage from the code that way (e.g. decompositions) :))

_Note: There is no need for you to push this logic now, let's just track it as TODO item, with a good second issues label :))_